### ISSUE TITLE
How is the progress now?

### ISSUE THREAD

**Issue by KKould** [iJustErikk/yadb#1]
Is the lsm storage completed? It seems that the roadmap is consistent with tanlentplan/tinysql

**Comment by iJustErikk** [comment]
Yeah its roughly done. Needs iterators, performance enhancements (can't scale with number of cores on writes yet) and statistics to back a good sql db (maybe mvcc/ssi if I want to implement transactions in the storage engine). Really only iterators needs to be implemented for a tiny sql db. 

I should check that out.

**Comment by KKould** [comment]
We are doing the same thing and hope to have the opportunity to communicate more :)

**Comment by iJustErikk** [comment]
Yeah, we could potentially merge efforts. Have you talked to Tyler Neely (sled)?

What are you interested in doing with storage? [b-epsilon trees](http://supertech.csail.mit.edu/papers/BenderFaJa15.pdf) are cool. Combining that optimization with modern b-tree techniques would be cool. 

Are you interested in DB stuff past storage/sql engines? I'd be interested in implementing Dynamo or something similar. 

Let's chat over whatsapp or linkedin, send me an email here: erikaww (at) umich (dot) edu

**Comment by KKould** [comment]
At present, I basically focus on SQL layer coding (now writing unique indexes). I am more interested in research on LSM, but I am very interested in BwTree. I am also more interested in the consensus protocol of the network layer. After completing [KipSQL](https://github.com/KipData/KipSQL), I will implement Raft with Storage and SQL

**Comment by iJustErikk** [comment]
I plan on doing similar things. My main plan is to finish the storage engine, then implement a single node SQL db and raft and maybe dynamo as well. 

I'd like to implement a causally consistent dynamo too. Are there any research algorithms you'd like to implement? Maybe for distributed transactions or crdts or compute?

**Comment by KKould** [comment]
Things other than SQL are still too far away for me. Now the most important thing is to pass KipSQL through TPCH and complete the index.
